src/sentry/incidents/logic.py

```diff
@@ -52,11 +62,74 @@ def create_incident(
             IncidentGroup.objects.bulk_create([
                 IncidentGroup(incident=incident, group=group) for group in groups
             ])
+
+        if status == IncidentStatus.CREATED:
+            activity_status = IncidentActivityType.CREATED
+        else:
+            activity_status = IncidentActivityType.DETECTED
+
+        event_stats_snapshot = create_initial_event_stats_snapshot(incident)
+        create_incident_activity(
+            incident,
+            activity_status,
+            event_stats_snapshot=event_stats_snapshot,
+        )
     return incident
 
 
-def build_incident_query_params(incident):
-    params = {'start': incident.date_started, 'end': incident.current_end_date}
+def create_initial_event_stats_snapshot(incident):
+    """
+    Creates an event snapshot representing the state at the beginning of
+    an incident. It's intended to capture the history of the events involved in
+    the incident, the spike and a short period of time after that.
+    """
+    initial_period_length = min(
+        timezone.now() - incident.date_started,
+        MAX_INITIAL_INCIDENT_PERIOD,
+    )
+    end = incident.date_started + initial_period_length
+    start = end - (initial_period_length * 8)
+    return create_event_stat_snapshot(incident, start, end)
+
+
+def create_incident_activity(
+    incident,
+    activity_type,
+    user=None,
+    value=None,
+    previous_value=None,
+    comment=None,
+    event_stats_snapshot=None,
+):
+    return IncidentActivity.objects.create(
+        incident=incident,
+        type=activity_type.value,
+        user=user,
+        value=value,
+        previous_value=previous_value,
```

Comment:
This might be unnecessary, since we can just look at the previous activity of the same type to get this value.